web/source/osk/preProcessor.ts

```diff
@@ -98,10 +98,38 @@ namespace com.keyman.osk {
           Lkc.source = touch;
           Lkc.keyDistribution = keyDistribution;
         }
-        return keyman.textProcessor.processKeyEvent(Lkc, e);
+
+        if(!keyman.isEmbedded) {
+          keyman.uiManager.setActivatingUI(true);
+          com.keyman.dom.DOMEventHandlers.states._IgnoreNextSelChange = 100;
+          keyman.domManager.focusLastActiveElement();
+          com.keyman.dom.DOMEventHandlers.states._IgnoreNextSelChange = 0;
+        }
+
+        let retVal = PreProcessor.handleClick(Lkc, e);
+
+        // Now that processing is done, we can do a bit of post-processing, too.
+        keyman.uiManager.setActivatingUI(false);	// I2498 - KeymanWeb OSK does not accept clicks in FF when using automatic UI
```

Comment:
Should this be wrapped in `if(!keyman.isEmbedded)` for consistency?

Reply:
I don't believe so, though this is a part of the OSK that I don't yet fully understand.  The check wasn't originally there, so I felt no need to add it.  (It's not worth risking a break, especially since I'm not 100% on its role.)

My impression is that this is the default value for the variable, so it's a safe call to make.  (No values should actually change.)  That said, I did notice that some touch handlers always set `true`, so there's a chance this actually cleans up after those handlers, even in embedded.